### ISSUE TITLE
binary_operator_spaces expanding lines each time it's ran against same file

### ISSUE THREAD

**Issue by jaydiablo** [FriendsOfPHP/PHP-CS-Fixer#6344]
## Bug report

I have a code snippet, that each time the `binary_operator_spaces` fixer is run, it keeps expanding the lines. This only happens in PHP CS Fixer versions greater than `3.6.0` (seems to do nothing to the code in that version, but `3.7.0` and `3.8.0` exhibit this behavior).

Here's the config for this particular fixer:

```
'binary_operator_spaces' => array(
            'default'   => 'align_single_space_minimal',
            'operators' => array('||' => null, '&&' => null)
        ),
```

Running on PHP 8.0.14.

### Code snippet that reproduces the problem

```php
<?php

$chunkStart = $page - (floor($chunk / 2));
$chunkEnd   = $page + (ceil($chunk / 2) - 1);
```

After the first run, the code changes to this:

```php
<?php

$chunkStart = $page                     - (floor($chunk / 2));
$chunkEnd   = $page + (ceil($chunk                      / 2) - 1);
```

If I run again:

```php
<?php

$chunkStart = $page                                          - (floor($chunk / 2));
$chunkEnd   = $page + (ceil($chunk                                           / 2) - 1);
```

It just keeps expanding the lines each time php-cs-fixer is run (as long as there's no cache for the file).

**Comment by jaydiablo** [comment]
Just FYI, this still happens with the latest version (3.9.2).

**Comment by VincentLanglet** [comment]
HI @jaydiablo

can you try with the master branch, this might be fixed by https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/2c36a47a04e4a11f7242a55e5b993a84a1f0af75

**Comment by julienfalque** [comment]
I confirm #6515 fixed this issue.

**Comment by VincentLanglet** [comment]
I dunno if https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5605 is fixed too, this seems similar but I'm not sure

**Comment by julienfalque** [comment]
Nope, #5605 is still failing.

**Comment by jaydiablo** [comment]
@VincentLanglet this appears to be resolved now (using `dev-master`, also in `3.10.0`).

Thanks!
